network/index.js

```diff
@@ -37,9 +37,10 @@ const useNetworkStatus = initialEffectiveConnectionType => {
   useEffect(() => {
     if (!unsupported) {
       const navigatorConnection = navigator.connection;
+      const networkAvailability = navigator.onLine;
```

Comment:
Perhaps you could add event listeners for the `online` and `offline` events as well, so this hook can stay updated as the user's network connectivity changes throughout their session?

Reply:
Yeah, sure. that would be a better way.